### ISSUE TITLE
Dart indentation issue 

### ISSUE THREAD

**Issue by SzelamC** [nvim-treesitter/nvim-treesitter#4087]
### Describe the bug

https://user-images.githubusercontent.com/78899518/210409812-f25f0d8f-2acd-4ef0-90b7-f6ca6fec45b3.mov

The indentation for dart seems have indentation when newline was insert after a curly bracket.

### To Reproduce

just install dart treesitter and press enter after curly bracket

### Expected behavior

just a newline without indentation

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.7 (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v18.12.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: Apple clang version 14.0.0 (clang-1400.0.29.202)
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "arm64",
  release = "21.6.0",
  sysname = "Darwin",
  version = "Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000"
} ~

Parser/Features         H L F I J
  - graphql             ✓ . . ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ . ✓
  - prisma              ✓ . . . .
  - rego                ✓ . . . ✓
  - scala               ✓ . ✓ . ✓
  - scheme              ✓ . ✓ . ✓
  - solidity            ✓ . . . .
  - sql                 ✓ . . . ✓
  - svelte              ✓ . ✓ ✓ ✓
  - swift               ✓ ✓ . . .
  - terraform           ✓ . ✓ ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - zig                 ✓ . ✓ ✓ ✓
  - dockerfile          ✓ . . . ✓
  - pascal              ✓ ✓ ✓ ✓ ✓
  - eex                 ✓ . . . ✓
  - vim                 ✓ ✓ ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - hack                ✓ . . . .
  - norg                . . . . .
  - vala                ✓ . ✓ . .
  - lalrpop             ✓ ✓ . . ✓
  - cooklang            ✓ . . . .
  - astro               ✓ ✓ ✓ ✓ ✓
  - teal                ✓ ✓ ✓ ✓ ✓
  - gitignore           ✓ . . . .
  - sxhkdrc             ✓ . ✓ . ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - bibtex              ✓ . ✓ ✓ .
  - ledger              ✓ . ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - vue                 ✓ . ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - bash                ✓ ✓ ✓ . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓
  - php                 ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - cuda                ✓ ✓ ✓ ✓ ✓
  - dart                ✓ ✓ . ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.0-dev-557+g94ce25065-dirty
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by szelc@Chans-MacBook-Pro.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/HEAD-94ce250/share/nvim"

Run :checkhealth for more info
```

**Comment by theHamsta** [comment]
C, Java and JS had a similar issue that got fixed by https://github.com/theHamsta/nvim-treesitter/blob/6e835755dc24208cdd92884ace91183e87a94c84/queries/c/indents.scm#L22 , https://github.com/theHamsta/nvim-treesitter/blob/6e835755dc24208cdd92884ace91183e87a94c84/queries/java/indents.scm#L27-L25 or https://github.com/theHamsta/nvim-treesitter/blob/6e835755dc24208cdd92884ace91183e87a94c84/queries/ecma/indents.scm#L39-L37

Could you check whether the same change would fix the issue for dart? You can you github.com/nvim-treesitter/playground or `:lua vim.treesitter.show_tree()` when you're on nightly to how the corresponding nodes are called in dart.

**Comment by SzelamC** [comment]
The change can fix the issue. I appreciate for your replying!!!!

**Comment by theHamsta** [comment]
@SzelamC we would really appreciate a PR. We should consider this issue only closed when everyone has the fix.

**Comment by amaanq** [comment]
Closed via #4167
